### PR TITLE
Add API validation for server pool counts

### PR DESCRIPTION
**Pull request by boingram**
Added checks for non-zero server pool size and non-zero min/max counts for server pools according to #214 

Docs to follow in subsequent PR

### CODE REVIEW

**Comment by kris-nova** [comment]
Can we fix the headers? I think you just need to run `make update-headers` and send another commit - then this should be good to merge.

File: cutil/initapi/validate.go [hunk 1]
```diff
@@ -14,4 +14,22 @@
 
 package initapi
 
-// keep
+import (
+	"fmt"
+	"github.com/kris-nova/kubicorn/apis/cluster"
+)
+
+func serverPoolCounts(initCluster *cluster.Cluster) error {
```
Comment (kris-nova) [inline cutil/initapi/validate.go]: Can we break these up into different functions and also give *all* the functions the prefix of `validate`?

For example we want a function called something like `validateServerPoolGreaterThan1` as well as `validateServerPoolMaxCountGreaterThan1` etc etc...

That way we can send another PR that will load these functions dynamically using reflection and we don't have to include the function name in the list anymore!

File: cutil/initapi/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	}
+	err := serverPoolCounts(c)
+	if err != nil {
+		t.Fatalf("error message incorrect for valid server pool counts"+
+			"should be: nil\n"+
+			"got:       %v\n", err)
+	}
+}
+
+func TestServerPoolCountsSad(t *testing.T) {
```
Comment (kris-nova) [inline cutil/initapi/validate_test.go]: Following the syntax above, can we have matching test functions for each of the validate functions and not lump them together? I know it seems like a lot of work, but once we have the validation system in place, the function name will serve as the message that tells us what went wrong 😉

File: cutil/initapi/validate.go (hunk 1 above)
Comment (boingram) [inline cutil/initapi/validate.go]: Certainly! How exactly would we use reflection to accomplish that? It all seems like dark magic to me, so a real-world example would be pretty helpful.

File: cutil/initapi/validate_test.go (hunk 2 above)
Comment (boingram) [inline cutil/initapi/validate_test.go]: Yep, changes on the way!

**Comment by boingram** [comment]
Updated to address comments. 

I was unable to get the make update-headers target to work for me due to sed complaining about an invalid command code, but I inserted the header manually, ran the check-headers script, and that passed

**Comment by kris-nova** [comment]
Gorgeous! Let's merge and we need to open an issue regarding the header bug you experienced. Do you have the error message?

**Comment by boingram** [comment]
Yep, thank god for having way too many iterm tabs still open

    $ make update-headers
    sed: 1: "./cutil/initapi/validat ...": invalid command code .

**Comment by boingram** [comment]
#228 created in reference to the header update script bug mentioned above

**Pull request merged**